internal/client.go

```diff
@@ -352,6 +352,25 @@ type (
 		// GetWorkerTaskReachability returns which versions are is still in use by open or closed workflows.
 		GetWorkerTaskReachability(ctx context.Context, options *GetWorkerTaskReachabilityOptions) (*WorkerTaskReachability, error)
 
+		// DescribeTaskQueueEnhanced returns information about the target task queue, broken down by Build Id:
+		//   - List of pollers
+		//   - Workflow Reachability status
+		//   - Backlog info for Workflow and/or Activity tasks
+		// When not supported by the server, it returns an empty [TaskQueueDescription]
+		// WARNING: Worker versioning is currently experimental, and requires server 1.24+
+		DescribeTaskQueueEnhanced(ctx context.Context, options *DescribeTaskQueueEnhancedOptions) (TaskQueueDescription, error)
```

Comment:
I know the old new version took pointers for options, I think @cretz prefers it without pointers. The SDK is **not** consistent here, but we should try to be consistent for new APIs. I think without a pointer is probably nicer since we don't have to deal with`nil`.

Reply:
Happy to change that, I saw 50/50 of APIs using pointers, and the old versioning always using it. 
But if we do change it, we should also change the new `UpdateWorkerVersioningRules` and `GetWorkerVersioningRules` to not have pointers, so that all are consistent...
@cretz what do you think?

Reply:
:+1: Confirmed on Slack, without pointers is best IMO